src/dynimage.rs

```diff
@@ -299,8 +299,8 @@ impl DynamicImage {
     }
 
     /// Return a cut out of this image delimited by the bounding rectangle.
-    pub fn crop(&mut self, x: u32, y: u32, width: u32, height: u32) -> DynamicImage {
-        dynamic_map!(*self, ref mut p => imageops::crop(p, x, y, width, height).to_image())
+    pub fn crop(&self, x: u32, y: u32, width: u32, height: u32) -> DynamicImage {
```

Comment:
AFAICT this will only allow code to compile that didn't previously (with an immutable `DynImage`) and add a warning (like below with `let mut intermediate = self.resize_exact(width2, height2, filter);`).

Will use for the same name as for the new function in `imageops`.

Reply:
No, it's definitely a breaking change. One could have used the method as a function pointer with the specific mutable signature.

```rust
const FN: fn(&mut DynamicImage, u32, u32, u32, u32) -> DynamicImage 
    = DynamicImage::crop;
```